crawler/src/main/java/com/movie/recommender/crawler/client/SerpApiClient.java

```diff
@@ -0,0 +1,44 @@
+package com.movie.recommender.crawler.client;
+
+import com.movie.recommender.common.model.MovieShowtimesResponse;
+import com.movie.recommender.lib.http.AuthProvider;
+import com.movie.recommender.lib.http.HttpClient;
+import lombok.extern.slf4j.Slf4j;
+import java.util.Optional;
+
+@Slf4j
+public class SerpApiClient {
```

Comment:
`callApi` and `getApiKey` methods, as well as the `httpClient` field, are nearly identical in `SerpApiClient` and `TmdbApiClient`
this leads to redundancy
create an abstract base class, `ApiClient`, that includes `callApi`, `getApiKey`, and `httpClient`
this will eliminate duplicate code
example
```
public class TmdbApiClient extends ApiClient {

    public TmdbApiClient() {
        super("TMDB_API_KEY");
    }
    
    // Other TmdbApiClient-specific methods...
}

public class SerpApiClient extends ApiClient {

    public SerpApiClient() {
        super("SERP_API_KEY");
    }
    
    // Other SerpApiClient-specific methods...
}
```